Server/src/services/authenticationService.ts

```diff
@@ -0,0 +1,142 @@
+            this.requestResponse.status = "Failure";
+            this.requestResponse.statusCode = 400;
+            this.requestResponse.response = "Email is already in the System. Please check again";
+
+            return this.requestResponse
+        }
+        this.requestResponse.status = "Success";
+        this.requestResponse.statusCode = 200;
+        return this.requestResponse
+
+    }
+
+    /**
+     * Resposible for authenticating user request for system access. Email is 
+     * first verified for existence, then the associated user password hash is 
+     * verified against the existence password hash in the database. Once a match 
+     * is established, the service will request the parameters required to build the
+     * JWT access token. After retrieval, a JWT token is built and sent back to the user
+     * @param userInformation - This params contains only UserName & PassWord
+     */
+    async checkLoginCredentials(userInformation: ILoginInformation): Promise<IResponse> {
+
+        let verifiedEmail: boolean;
+        let savedPasswordHash: string;
+        let accessToken: string;
+
+        try {
+            verifiedEmail = await AuthenticationModel.verifyIfEmailExists(userInformation.email);
+            if (!verifiedEmail)
+                throw new Error();
+        }
+        catch (error) {
+            this.requestResponse.status = "Failure";
+            this.requestResponse.statusCode = 400;
+            this.requestResponse.response = "Email is not in the System or its mispelled. Check Again";
+
+            return this.requestResponse
+        }
+
+        savedPasswordHash = await AuthenticationModel.verifyPassword(userInformation.email);
+
+        if (!(await argon2.verify(savedPasswordHash, userInformation.password))) {
+            this.requestResponse.status = "Failure";
+            this.requestResponse.statusCode = 400;
+            this.requestResponse.response = "Password does not match";
+
+            return this.requestResponse
+        }
+        else {
+            let jwtParams: IJwtParams;
+            try {
+                jwtParams = await AuthenticationModel.obtainJWTParams(userInformation.email);
+                accessToken = await this.generateJwtToken(jwtParams);
+            } catch (error) {
+                this.requestResponse.status = "Failure";
+                this.requestResponse.statusCode = 500;
+                this.requestResponse.response = "Internal Server Issue. Please try again later";
+                return this.requestResponse
+            }
+
+            this.requestResponse.status = "Success";
+            this.requestResponse.statusCode = 200;
+            this.requestResponse.response = accessToken;
+        }
+
+        return this.requestResponse;
+    }
+    /**
+     * Resposible for hashing the user input password
+     * @param password - user password
+     */
+    private async hashPassword(userPassword: string): Promise<string> {
```

Comment:
its not safe to send a password in the request since its interceptable, backend should never know the actual password, hashing should be done client side

Reply:
This is not a good idea ever.
- https://en.wikipedia.org/wiki/Pass_the_hash

Reply:
yeah, i admit it was dumb to hash client side, i think i learnt this from my soen 287 class lol